realtime/redisPublisher.js

```diff
@@ -92,32 +92,49 @@ function publishObject(inst, event, changedKeys, ignoreAttributes) {
  * a absolutePath field added to it before the sample is published to the redis
  * channel.
  * @param  {Object} sampleInst - The sample instance to be published
- * @param  {Model} model - The subject model used to get the related subject
- * instance
+ * @param  {Model} subjectModel - The subject model to get the related
+ * subject instance
  * @param  {String} event  - Type of the event that is being published
+ * @param  {Model} aspectModel  - The aspect model to get the related
+ * aspect instance
  * @returns {Promise} - which resolves to a sample object
  */
-function publishSample(sampleInst, model, event) {
+function publishSample(sampleInst, subjectModel, event, aspectModel) {
   const eventType = event || getSampleEventType(sampleInst);
   const sample = sampleInst.get ? sampleInst.get() : sampleInst;
-  const subName = sample.name.split('|')[0];
-  const options = {};
-  options.where = { absolutePath: subName };
-  return model.findOne(options)
+  const nameParts = sample.name.split('|');
+  const subName = nameParts[0];
+  const aspName = nameParts[1];
+  const subOpts = {
+    where: {
+      absolutePath: subName,
+    },
+  };
+  const aspOpts = {
+    where: {
+      name: aspName,
```

Comment:
aspectName and subjectName are always from the "sample.name" field from the db and the "sample.name" field is formed from its subject and its aspect in the db. So the case always matches when we do the find here.